### PR TITLE
[dispatcher] refactor dispatcher to simplify adding new message

### PR DESCRIPTION
**Pull request by envestcc**
# Description

The main purpose of refactoring the dispatcher is to make it easier to add more messages. Specific changes include: 
- Extracting the `msg_queue` component to manage the queue worker and message mappping
- Using a unify `message` structure to simplify the processing of different types of messages

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: dispatcher/dispatcherV2.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+	consensusQueue := make(chan *broadcastMsg, 100)
+	actionQueue := make(chan *broadcastMsg, 100)
+	blockQueue := make(chan *broadcastMsg, 100)
+	actionQueueUnicast := make(chan *unicastMsg, 100)
+	d.broadcastQueueMap[iotexrpc.MessageType_BLOCK] = blockQueue
+	d.broadcastQueueMap[iotexrpc.MessageType_CONSENSUS] = consensusQueue
+	d.broadcastQueueMap[iotexrpc.MessageType_ACTION] = actionQueue
+
+	for i := 0; i < 10; i++ {
+		d.wg.Add(1)
```
Comment (CoderZhi) [inline dispatcher/dispatcherV2.go:57]: where will the `d.wg.WaitAll()` be called?

File: dispatcher/dispatcherV2.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+	}
+	unicastMsg struct {
+		ctx     context.Context
+		msg     proto.Message
+		chainID uint32
+		peer    peer.AddrInfo
+	}
+)
+
+func (d *dispatcherV2) ConfigureMessage() {
```
Comment (CoderZhi) [inline dispatcher/dispatcherV2.go:44]: start?

File: dispatcher/dispatcher.go [hunk 3]
```diff
@@ -37,24 +40,16 @@ type (
...
 	// DefaultConfig is the default config
 	DefaultConfig = Config{
-		ActionChanSize:             5000,
-		BlockChanSize:              1000,
-		BlockSyncChanSize:          400,
+		ActionChanSize:    5000,
+		BlockChanSize:     1000,
+		BlockSyncChanSize: 400,
+		ConsensusChanSize: 1000,
+		MiscChanSize:      1000,
```
Comment (dustinxie) [inline dispatcher/dispatcher.go:47]: 4844 plan to use the miscChan? if that's the case, consider just use the feature name like `blobChan`?

File: dispatcher/dispatcherV2.go [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+	"github.com/libp2p/go-libp2p-core/peer"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/iotexproject/iotex-core/pkg/lifecycle"
+	"github.com/iotexproject/iotex-core/pkg/log"
+)
+
+type (
+	dispatcherV2 struct {
```
Comment (dustinxie) [inline dispatcher/dispatcherV2.go:20]: adding `msgQueueMgr` is a good move, to handle the message dispatch into different queue 
on top of that, i think you can directly modify current dispatcher, no need to add a V2, so we can clearly see the changes

File: dispatcher/dispatcher.go (hunk 3 above)
Comment (envestcc) [reply dispatcher/dispatcher.go:47]: I think blobTx can use actionChan

File: dispatcher/dispatcherV2.go [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+		if err := subscriber.HandleNodeInfo(message.ctx, message.peer, msg); err != nil {
+			log.L().Warn("Failed to handle node info message.", zap.Error(err))
+		}
+	default:
+		msgType, _ := goproto.GetTypeFromRPCMsg(message.msg)
+		log.L().Warn("Unexpected msgType handled by HandleBroadcast.", zap.Any("msgType", msgType))
+	}
+}
+
+func (d *dispatcherV2) dispatchUnicastMsg(message *message) {
```
Comment (dustinxie) [inline dispatcher/dispatcherV2.go:210]: the logic actually same as `dispatchBroadcastMsg`, can combine into one func

File: dispatcher/dispatcherV2.go [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+func (d *dispatcherV2) updateMetrics(msg *message, queue chan *message) {
+	subscriber := d.subscriber(msg.chainID)
+	if subscriber != nil {
+		subscriber.ReportFullness(msg.ctx, msg.msgType, float32(len(queue))/float32(cap(queue)))
+	}
+}
+
+func (d *dispatcherV2) dispatchBroadcastMsg(message *message) {
+	if !d.IsReady() {
```
Comment (dustinxie) [inline dispatcher/dispatcherV2.go:171]: move this check to before dispatch

File: dispatcher/msg_queue.go [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	return nil
+}
+
+func (m *msgQueueMgr) Stop() error {
+	close(m.quit)
+	m.wg.Wait()
+	return nil
+}
+
+func (m *msgQueueMgr) comsume(q string) {
```
[the remark this answers is not in the document]
Comment (envestcc) [reply dispatcher/msg_queue.go]: fixed

File: dispatcher/dispatcherV2.go (hunk 4 above)
Comment (envestcc) [reply dispatcher/dispatcherV2.go]: combined the v2 with v1

File: dispatcher/dispatcherV2.go (hunk 5 above)
Comment (envestcc) [reply dispatcher/dispatcherV2.go]: conbined

File: dispatcher/dispatcherV2.go (hunk 6 above)
Comment (envestcc) [reply dispatcher/dispatcherV2.go]: fixed

**Review by dustinxie — APPROVED** [review]

**Review by CoderZhi — APPROVED** [review]

**Pull request merged**